src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -693,3 +697,79 @@ class StatementSegment(BaseSegment):
             Ref("DropModelStatementSegment"),
         ],
     )
+
+
+@spark3_dialect.segment(replace=True)
+class JoinClauseSegment(BaseSegment):
+    """Any number of join clauses, including the `JOIN` keyword.
+
+    https://spark.apache.org/docs/3.0.0/sql-ref-syntax-qry-select-join.html
+    TODO: Add NATURAL JOIN syntax.
+    """
+
+    type = "join_clause"
+    match_grammar = Sequence(
+        # NB These qualifiers are optional
+        # TODO: Allow nested joins like:
+        # ....FROM S1.T1 t1 LEFT JOIN ( S2.T2 t2 JOIN S3.T3 t3 ON t2.col1=t3.col1) ON tab1.col1 = tab2.col1
+        OneOf(
+            "CROSS",
+            "INNER",
+            Sequence(
+                OneOf(
+                    "FULL",
+                    "LEFT",
+                    "RIGHT",
+                ),
+                Ref.keyword("OUTER", optional=True),
+            ),
+            Sequence(
+                Ref.keyword("LEFT", optional=True),
+                "SEMI",
+            ),
+            Sequence(
+                Ref.keyword("LEFT", optional=True),
+                "ANTI",
+            ),
+            optional=True,
+        ),
+        Ref("JoinKeywords"),
+        Indent,
+        Sequence(
+            Ref("FromExpressionElementSegment"),
+            Conditional(Dedent, indented_using_on=False),
+            # NB: this is optional
+            OneOf(
+                # ON clause
+                Ref("JoinOnConditionSegment"),
+                # USING clause
+                Sequence(
+                    "USING",
+                    Indent,
+                    Bracketed(
+                        # NB: We don't use BracketedColumnReferenceListGrammar
+                        # here because we're just using SingleIdentifierGrammar,
+                        # rather than ObjectReferenceSegment or ColumnReferenceSegment.
+                        # This is a) so that we don't lint it as a reference and
+                        # b) because the column will probably be returned anyway
+                        # during parsing.
+                        Delimited(
+                            Ref("SingleIdentifierGrammar"),
+                            ephemeral_name="UsingClauseContents",
+                        )
+                    ),
+                    Dedent,
+                ),
+                # Unqualified joins *are* allowed. They just might not
+                # be a good idea.
+                optional=True,
+            ),
+            Conditional(Indent, indented_using_on=False),
+        ),
+        Dedent,
+    )
+
+    def get_eventual_alias(self) -> AliasInfo:
```

Comment:
If you copy the rule from the ansi dialect segment instead of writing it again, the codecov complaint will go away. It's better to do that anyway as you're not actually changing the functionality

Reply:
@WittierDinosaur Thanks, I've made the change to get `get_eventual_alias` from the ansi dialect and seems to have satisfied codecov 👍